### PR TITLE
Add LAP SE 1.8 as a release build env

### PR DESCRIPTION
**Pull request by keadyk**
### Background

* Codifying changes made during draco-7_12 release process :)

### Purpose of Pull Request

* Add LAP SE1.8 environments as release builds for CTS/ATS-1
* (This seems useful given that LAP is now the default draco env on those machines.)
* [Fixes re-git issue #1363](https://re-git.lanl.gov/draco/draco/-/issues/1363)

### Description of changes

* Add functions to bash scripts to load LAP SE environments
* Small changes have already been made to corresponding draco module files to ensure the correct mpich is loaded.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash3/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [ ] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

File: tools/ats1-env.sh [hunk 1]
```diff
@@ -112,6 +112,50 @@ case "$ddir" in
...
+
+      sysname=$(/usr/projects/hpcsoft/utilities/bin/sys_name)
+      module use --append "/usr/projects/draco/Modules/${sysname}"
+
+      run "module unload draco"
+      run "module unload PrgEnv-cray"
+      run "module load PrgEnv-intel"
+      run "module load lapse/1.8-intel"
+      # Be explicit about unloading haswell / loading the knl module!
+      # Something went awry when I tried to use the intel1904env-knl logic.
```
Comment (keadyk) [inline tools/ats1-env.sh]: @KineticTheory Does this section seem okay? I originally followed your logic from the standard intel-19 module, but something went awry during my first attempt to release. The `module swap` didn't work properly, so the knl tag got appended to the build dir and the partition was set to knl... but the haswell module was still loaded.

This more explicit unload/load version worked, so I went with it. :)

File: tools/ats1-env.sh [hunk 2]
```diff
@@ -24,7 +24,7 @@ else
 fi
 
 # The following toolchains will be used when releasing code
-environments="cce11env intel1904env intel1904env-knl"
+environments="cce11env intel1904env lapse18intelenv intel1904env-knl lapse18intelenv-knl"
```
Comment (KineticTheory) [inline tools/ats1-env.sh]: I would remove the `intel1904env` entry here and the function definition.

File: tools/ats1-env.sh [hunk 3]
```diff
@@ -112,6 +112,50 @@ case "$ddir" in
...
+
+    function lapse18intelenv()
+    {
+      unset partition
+      unset jobnameext
+
+      sysname=$(/usr/projects/hpcsoft/utilities/bin/sys_name)
+      module use --append "/usr/projects/draco/Modules/${sysname}"
+
+      run "module unload draco"
```
Comment (KineticTheory) [inline tools/ats1-env.sh]: `unload draco lapse`

File: tools/ats1-env.sh [hunk 4]
```diff
@@ -112,6 +112,50 @@ case "$ddir" in
...
+      run "module unload PrgEnv-cray"
```
Comment (KineticTheory) [inline tools/ats1-env.sh]: You can do something like this to be more generic:.  
`module unload PrgEnv-cray PrgEnv-intel PrgEnv-gnu`

File: tools/ats1-env.sh (hunk 1 above)
Comment (KineticTheory) [reply tools/ats1-env.sh]: See my comments above. I think the `unload` commands need to cover more possibilities.

**Review by KineticTheory — APPROVED** [review]

File: tools/ats1-env.sh (hunk 4 above)
Comment (keadyk) [reply tools/ats1-env.sh]: Ok, thanks! :)

File: tools/ats1-env.sh (hunk 2 above)
Comment (keadyk) [reply tools/ats1-env.sh]: Ok, sounds good. I wasn't super keen on the ballooning number of environments I was creating :)

**Comment by keadyk** [comment]
@KineticTheory I addressed your comments - thanks! I tested the env commands manually and everything is functioning. This is ready to merge when you are!

**Pull request merged**